spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/ConversionUtils.java

```diff
@@ -17,234 +17,34 @@
-					cal.get(Calendar.DAY_OF_MONTH));
-		}
-	};
-
-	public static final Converter<com.google.cloud.Date, Date> SPANNER_TO_JAVA_DATE_CONVERTER =
-			new Converter<com.google.cloud.Date, Date>() {
-		@Nullable
-		@Override
-		public Date convert(com.google.cloud.Date date) {
-			Calendar cal = Calendar.getInstance();
-			cal.set(date.getYear(), date.getMonth() - 1, date.getDayOfMonth());
-			return cal.getTime();
-		}
-	};
-
-	public static final Converter<Instant, Timestamp> INSTANT_TIMESTAMP_CONVERTER =
-			new Converter<Instant, Timestamp>() {
-		@Nullable
-		@Override
-		public Timestamp convert(Instant instant) {
-			return Timestamp.ofTimeSecondsAndNanos(instant.getEpochSecond(), instant.getNano());
-		}
-	};
-
-	public static final Converter<Timestamp, Instant> TIMESTAMP_INSTANT_CONVERTER =
-			new Converter<Timestamp, Instant>() {
-		@Nullable
-		@Override
-		public Instant convert(Timestamp timestamp) {
-			return Instant.ofEpochSecond(timestamp.getSeconds(), timestamp.getNanos());
-		}
-	};
-
-	public static final Converter<java.sql.Timestamp, Timestamp> JAVA_TO_SPANNER_TIMESTAMP_CONVERTER =
-			new Converter<java.sql.Timestamp, Timestamp>() {
-		@Nullable
-		@Override
-		public Timestamp convert(java.sql.Timestamp timestamp) {
-			return Timestamp.of(timestamp);
-		}
-	};
-
-	public static final Converter<Timestamp, java.sql.Timestamp> SPANNER_TO_JAVA_TIMESTAMP_CONVERTER =
-			new Converter<Timestamp, java.sql.Timestamp>() {
-		@Nullable
-		@Override
-		public java.sql.Timestamp convert(Timestamp timestamp) {
-			return java.sql.Timestamp.from(TIMESTAMP_INSTANT_CONVERTER.convert(timestamp));
-		}
-	};
-
-	public static final Converter<byte[], ByteArray> JAVA_TO_SPANNER_BYTE_ARRAY_CONVERTER =
-			new Converter<byte[], ByteArray>() {
-		@Nullable
-		@Override
-		public ByteArray convert(byte[] bytes) {
-			return ByteArray.copyFrom(bytes);
-		}
-	};
-
-	public static final Converter<ByteArray, byte[]> SPANNER_TO_JAVA_BYTE_ARRAY_CONVERTER =
-			new Converter<ByteArray, byte[]>() {
-		@Nullable
-		@Override
-		public byte[] convert(ByteArray bytes) {
-			return bytes.toByteArray();
-		}
-	};
-
-	/**
-	 * Converters from common types to those used by Spanner.
-	 */
-	public static final Collection<Converter> DEFAULT_SPANNER_WRITE_CONVERTERS = ImmutableSet
-			.of(JAVA_TO_SPANNER_DATE_CONVERTER, INSTANT_TIMESTAMP_CONVERTER, JAVA_TO_SPANNER_BYTE_ARRAY_CONVERTER,
-					JAVA_TO_SPANNER_TIMESTAMP_CONVERTER, JAVA_SQL_TO_SPANNER_DATE_CONVERTER);
-
-	/**
-	 * Converters from common types to those used by Spanner.
-	 */
-	public static final Collection<Converter> DEFAULT_SPANNER_READ_CONVERTERS = ImmutableSet
-			.of(SPANNER_TO_JAVA_DATE_CONVERTER, TIMESTAMP_INSTANT_CONVERTER, SPANNER_TO_JAVA_BYTE_ARRAY_CONVERTER,
-					SPANNER_TO_JAVA_TIMESTAMP_CONVERTER, SPANNER_TO_JAVA_SQL_DATE_CONVERTER);
-
-	public static Class boxIfNeeded(Class propertyType) {
+	public static <T> Class<T> boxIfNeeded(Class<T> propertyType) {
```

Comment:
I don't think the return type and param type here should both be `Class<T>`. That's saying the original type and the boxed type should be exactly the same  type. That seems to go against the purpose of this class because we are supposed to be giving back a different type than what is received. I think this only works because any type that does need boxing can be safely cast as its boxed type like `Integer` can be cast as `int`. I don't doubt that casting like this can always be safe, but I think it just makes the definition of this function strange and self-contradicting .

I see you need this type parameter here on this line in AbstractSpannerCustomConverter:
`Class<T> boxedTargetType = ConversionUtils.boxIfNeeded(targetType);`. 
I think you should just be casting to `Class<T>` on this line instead of in `boxIfNeeded`.

Reply:
@balopat otherwise it looks good to me!